consensus/service/src/tx_manager.rs

```diff
@@ -293,17 +307,16 @@ impl<E: ConsensusEnclaveProxy, L: Ledger, UI: UntrustedInterfaces> TxManager<E,
             .combine(&tx_contexts, MAX_TRANSACTIONS_PER_BLOCK)
     }
 
-    /// A "shim" that converts the output of consensus into something that can be written to the ledger.
-    pub fn tx_hashes_to_block(
+    /// Forms a Block containing the transactions that correspond to the given hashes.
+    fn tx_hashes_to_block(
         &self,
         tx_hashes: &[TxHash],
+        parent_block: &Block,
```

Comment:
This now takes `parent_block`, which means that TxManager no longer needs to keep an instance of the ledger.